File: Problems/0141_Linked_List_Cycle/Project_CS/Operate_ListNode.cs
using System;

public class Operate_ListNode
{
    public ListNode set_nodes(int[] nums)
    {
        return set_nodes(nums, 0);
    }

    public ListNode set_nodes(int[] nums, int index)
    {
        if (nums == null)
            return null;

        if (index >= nums.Length)
            return null;
        
        ListNode node = new ListNode(nums[index]);
        node.next = set_nodes(nums, index + 1);

        return node;
    }

    public string output_nodes(ListNode node)
    {
        if (node == null)
            return "";

        string resultStr = node.val.ToString();

        if (node.next != null)
            resultStr += " -> " + output_nodes(node.next);

        return (resultStr);
    }
}

File: Problems/0001_0099/0002_Add_Two_Number/Project_C/lib/operate_listnode.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>
#include <math.h>

#include "../include/listnode.h"
#include "../include/operate_listnode.h"

struct ListNode* set_nodes(int nums[], int nums_Size, int index)
{
    if (index >= nums_Size)
        return NULL;
    
    struct ListNode* node = (struct ListNode*)malloc(sizeof(struct ListNode));

    node->val = nums[index];
    node->next = set_nodes(nums, nums_Size, index + 1);

    return node;
}


char* output_nodes(struct ListNode* ll)
{
    #define resultStr_MAX_SIZE  1024

    char* resultStr = (char *)malloc(sizeof(char)*resultStr_MAX_SIZE);
    sprintf(resultStr, "%d", ll->val); 

    if (ll->next != NULL) {
        strcat(resultStr, " -> ");
        strcat(resultStr, output_nodes(ll->next));
    }

    return resultStr;
}

int ListNode_free(struct ListNode* ll)
{
    if (ll == NULL)
        return 0;
    
    int result = 0;

    if (ll->next != NULL) {
        result += ListNode_free(ll->next);
    }

    free(ll);
    result++;

    return result;
}
